Detectors/FOCAL/simulation/src/Detector.cxx

```diff
@@ -165,6 +184,54 @@ Bool_t Detector::ProcessHits(FairVolume* v)
     }
   }
 
+  // Processing ECAL hits
+  if (TVirtualMC::GetMC()->CurrentMedium() == mMedSensECal) {
```

Comment:
Please make sure that pad and pixel layers are declared as separate sensitive volumes. Because the pixels need a different hit simulation (mixture of Detector simulation in ITSMFT and the gaussian diffusion model)

Reply:
Pixels and Pads are separated in geometry and in process hit method.